### PR TITLE
[Fix #903] Fix deprecation warning in LogSubscriber in Rails 7.1

### LINKED ISSUE #903: Warning when upgrading to Rails 7.1
## Expected behavior

I'm upgrading my app to `Rails 7.1` and I have a deprecation warning message that seems to be triggered by chewy.

## Actual behavior

This warning is displayed when running my testing suite:

```
DEPRECATION WARNING: Bolding log text with a positional boolean is deprecated and will be removed in Rails 7.2. Use an option hash instead (eg. `color("my text", :red, bold: true)`).
```

## Steps to reproduce the problem

The warning is triggered by the line located ([here](https://github.com/toptal/chewy/blob/22ead9a21b0908de856e0452dc7273db8c6725de/lib/chewy/log_subscriber.rb#L27))

As recommended by the warning message, the warning is not displayed if we move the line to:

```ruby
action = color(action, GREEN, bold: true)
```

Regarding the doc, the support is supported until `Rails 6.1`, is there any plan for the newly Rails release?

I'm taking the opportunity of this message to thank you for this gem.

## Version Information

* Chewy version 7.3.4
* Elasticsearch version
* Ruby version 3.2.2
* Rails version 7.1.1

### PR DESCRIPTION
**Pull request by alejandroperea**
Fixes https://github.com/toptal/chewy/issues/903

`color` method was changed in Rails 7.1, introducing a `mode_options` hash instead of a position parameter just for bold option.

This PR checks the ActiveSupport version and calls `color` method with the proper parameters.

Also, I've added the configuration for executing the CI for Rails 7.1 version

-----------------

Before submitting the PR make sure the following are checked:

* [ ] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [ ] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the changelog if the new code introduces user-observable changes. See [changelog entry format](https://github.com/toptal/chewy/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by konalegi** [comment]
Great, thank you for the PR, could you please update the change log as well? Thank you! Something like this https://github.com/toptal/chewy/pull/892/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR7. Once CI pass, I guess I'm fine with merge.

**Comment by alejandroperea** [comment]
@konalegi I've squashed the changes with the `CHANGELOG`.

The 7.1 builds were failing because of this:

https://github.com/toptal/chewy/pull/907/files#diff-de55eb9d876f7c9af5c2ed3ad73de1336e1ceadd73284015bfccf5d435e58de3R3

As it was unable to require `active_support/deprecation`. I don't know if this is OK for now (as chewy it is not using any deprecation). Or maybe is it better to remove it and rethink it when chewy needs to add a deprecation.

Let me know what you think 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ### New Features
 
+* [#888](https://github.com/toptal/chewy/pull/907): Fix deprecation warning in LogSubscriber for Rails 7.1 ([@alejandroperea](https://github.com/alejandroperea))
```
Comment (konalegi) [inline CHANGELOG.md:7]:
```suggestion
* [#907](https://github.com/toptal/chewy/pull/907): Fix deprecation warning in LogSubscriber for Rails 7.1 ([@alejandroperea](https://github.com/alejandroperea))
```

File: lib/chewy.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'active_support/version'
 require 'active_support/concern'
-require 'active_support/deprecation'
+require 'active_support/deprecation' if ActiveSupport.version < '7.1'
```
Comment (konalegi) [inline lib/chewy.rb:3]: You cannot compare like this, should be something 
```suggestion
require 'active_support/deprecation' if ActiveSupport.version < Gem::Version.new('7.1')
```

Comment (alejandroperea) [reply lib/chewy.rb]: That's completely true 🤦

**Comment by alejandroperea** [comment]
@konalegi Changes done 👍  I think you have to trigger the workflows. Let's see what happens

**Comment by alejandroperea** [comment]
@konalegi I was looking for the same error in other gems. Based on what [CocoaPods did](https://github.com/CocoaPods/CocoaPods/pull/12082) and the [official documentation](https://guides.rubyonrails.org/active_support_core_extensions.html#stand-alone-active-support), I believe requiring the bare minimum with `require 'active_suport'` and then the extensions needed, it fixes the problem.

**Comment by konalegi** [comment]
@alejandroperea I'll take a look, I don't know what exactly adds `require 'active_support'` don't really want to include some extra stuff, that we don't really need. I'll take a look in 4-5 hours.

**Comment by alejandroperea** [comment]
Sure @konalegi , no hurries! What I understand from the documentation, it only loads the bare minimum and some entry points for being able to load then the needed extensions, but not 100% sure.

Now the tests are passing (only one failed for Ruby 2.6 / Rails 6.1, not sure why). Let me know if you need further changes or investigation 👍

File: lib/chewy.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+require 'active_support'
```
Comment (konalegi) [inline lib/chewy.rb:1]: I confirm, it's pretty safe https://guides.rubyonrails.org/active_support_core_extensions.html#stand-alone-active-support

**Review by konalegi — APPROVED** [review]
Nice, thank you!

**Comment by konalegi** [comment]
I'll try to gather 2nd review, if not, will merge later.

**Comment by alejandroperea** [comment]
Thanks for checking it @konalegi !

**Comment by alejandroperea** [comment]
Thank you so much @konalegi !

**Pull request merged**